### PR TITLE
use fixed alias for cloudfront domain

### PR DESCRIPTION
**Pull request by bdehri**

### CODE REVIEW

File: controllers/awscluster_controller.go [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+		cm)
+	if err != nil {
+		logger.Error(err, "Failed to get the cluster-values configmap for cluster")
+		return ctrl.Result{}, err
+	}
+
+	if !controllerutil.ContainsFinalizer(cm, key.FinalizerName(iam.IRSARole)) {
+		patchHelper, err := patch.NewHelper(cm, r.Client)
+		if err != nil {
+			return ctrl.Result{}, err
```
Comment (fiunchinho) [inline controllers/awscluster_controller.go:122]: Using this function, we can get the stack trace when an error is returned, making it easier to know where the error happened
```
errors.WithStack(err)
```

File: controllers/awscluster_controller.go [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+		logger.Error(err, "Failed to get aws client session", "cluster_name", awsCluster)
+		return ctrl.Result{}, err
+	}
+
+	var iamService *iam.IAMService
+	{
+		c := iam.IAMServiceConfig{
+			AWSSession:                awsClientSession,
+			ClusterName:               awsCluster.Name,
+			MainRoleName:              "-",
```
Comment (fiunchinho) [inline controllers/awscluster_controller.go:164]: I see that the `MainRoleName` is `-`. Is this expected?

File: controllers/awscluster_controller.go [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+
+	if controllerutil.ContainsFinalizer(awsCluster, key.FinalizerName(iam.IRSARole)) {
+		patchHelper, err := patch.NewHelper(awsCluster, r.Client)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		controllerutil.RemoveFinalizer(awsCluster, key.FinalizerName(iam.IRSARole))
+		err = patchHelper.Patch(ctx, awsCluster)
+		if err != nil {
+			logger.Error(err, "failed to remove finalizer on awsCluster")
```
Comment (fiunchinho) [inline controllers/awscluster_controller.go:221]:
```suggestion
			logger.Error(err, "failed to remove finalizer from awsCluster")
```

File: controllers/awscluster_controller.go [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+const (
+	IRSASecretSuffix = "irsa-cloudfront"
+)
+
+// AWSClusterReconciler reconciles a Secret object
+type AWSClusterReconciler struct {
+	client.Client
+	EnableIRSARole            bool
+	Log                       logr.Logger
+	Scheme                    *runtime.Scheme
```
Comment (fiunchinho) [inline controllers/awscluster_controller.go:52]: I think you don't need this field and it could be removed.

**Review by fiunchinho — COMMENTED** [review]
I left some comments and suggestions. Let me know if you have any questions or want to talk about it.

**Comment by fiunchinho** [comment]
Please, add an entry to the changelog explaining the changes.

**Review by fiunchinho — APPROVED** [review]

File: controllers/awscluster_controller.go [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+)
+
+const (
+	IRSASecretSuffix = "irsa-cloudfront"
+)
+
+// AWSClusterReconciler reconciles a Secret object
+type AWSClusterReconciler struct {
+	client.Client
+	EnableIRSARole            bool
```
Comment (fiunchinho) [inline controllers/awscluster_controller.go:50]: I don't think this is used anymore.

File: controllers/awscluster_controller.go [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+	return r.reconcileNormal(ctx, logger, awsCluster)
+}
+
+func (r *AWSClusterReconciler) reconcileNormal(ctx context.Context, logger logr.Logger, awsCluster *capa.AWSCluster) (ctrl.Result, error) {
+	logger.Info("reconcile normal")
+	// add finalizer to AWSCluster
+	if !controllerutil.ContainsFinalizer(awsCluster, key.FinalizerName(iam.IRSARole)) {
+		patchHelper, err := patch.NewHelper(awsCluster, r.Client)
+		if err != nil {
+			return ctrl.Result{}, err
```
Comment (fiunchinho) [inline controllers/awscluster_controller.go:83]:
```suggestion
			return ctrl.Result{}, errors.WithStack(err)
```

File: controllers/awsmachinetemplate_controller.go [hunk 7]
```diff
@@ -204,31 +188,31 @@ func (r *AWSMachineTemplateReconciler) Reconcile(ctx context.Context, req ctrl.R
...
 				Namespace: req.NamespacedName.Namespace,
-				Name:      fmt.Sprintf("%s-%s", clusterName, IRSASecretSuffix),
+				Name:      fmt.Sprintf("%s-%s", clusterName, "cluster-values"),
 			},
-			secret)
+			cm)
 		if err != nil {
-			logger.Error(err, "Failed to get the irsa-cloudfront secret for cluster")
+			logger.Error(err, "Failed to get the cluster-values configmap for cluster")
 			return ctrl.Result{}, err
```
Comment (fiunchinho) [inline controllers/awsmachinetemplate_controller.go:201]:
```suggestion
			return ctrl.Result{}, errors.WithStack(err)
```

**Review by fiunchinho — CHANGES_REQUESTED** [review]
We could use `errors.WithStack(err)` in many more places to help debugging when something goes wrong.

**Comment by bdehri** [comment]
> We could use `errors.WithStack(err)` in many more places to help debugging when something goes wrong.

Every return statement from a reconciler should return error with stack now.

**Pull request merged**